### DISCUSSION TITLE
opus encoder configuration (exotic options)

### DISCUSSION THREAD

**Discussion by Hub-O-Gits** [enzo1982/freac#369] [Help & Support]
I'm not expert in digital audio, so some of this may be rudimentary, but there are two options in the opus encoder configuration that I haven't seen elsewhere, and I can't find documentation regarding their implementation:

- "Bandwidth"
- "Enable discontinuous transmission"

I was able to cobble together a general understanding of what they apparently refer to, but I don't see them referenced as settings in libopus or opusenc.exe, so I'm wondering about the following:

1. What's providing those options? (Switches, library calls, whatever.)
2. What affect do they have in detail?
3. Would "Fullband" be the default if you hadn't included a "Bandwidth" setting?

Thank you

**Comment by enzo1982** [comment]
These are regular options provided by the Opus codec library.

The bandwidth setting controls the bandpass filter applied by the encoder. The meaning of the values is:

 - Narrowband - 4 kHz
 - Mediumband - 6 kHz
 - Wideband - 8 kHz
 - Superwideband - 12 kHz
 - Fullband - 20 kHz

Setting this to anything less than fullband makes sense mostly when encoding speech. For music, fullband should be used in almost all cases.

The discontinuous transmission option enables the encoder to override the bitrate target and significantly reduce the bitrate (to almost 0) when encoding silence. This can, for example, reduce the overall bitrate by ~5% when encoding audio books, without affecting quality.
